californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseLayer.java

```diff
@@ -881,7 +895,7 @@ private void handleBlock2Response(final Exchange exchange, final Response respon
 								status.getBlockCount());
 						Response assembled = new Response(response.getCode());
 						status.assembleReceivedMessage(assembled);
-
+						assembled.getOptions().setBlock2(response.getOptions().getBlock2());
```

Comment:
Why we need to add this block2 option on the assembled one ?

Reply:
I added it, because the "plugchecker" assumes it. 
FMPOV, test- and analysis-clients benefit from that information.
On the other side, I don't think others are mixed up by not having pure "transparent blockwise".
Alternatively, we can adjust the test to not use "transparent mode", but this would require much more work.

Reply:
I feel strange that the "transparent mode" returns a complete response with block parameter :/

I understand this is the simple way to fix plugchecker, I just played a bit with it and I'm not sure to see the purpose of this plugchecker ? Do we still need it ? What does it bring more than our unit tests ?

Reply:
Yes :-).

1. it covers the interoperability test defined for 
    "ETSI IoT CoAP Plugtests, London, UK, 7--9 Mar 2014."
    (OK, the cf-plugtest-client does it similar)

2. it has detected the issues with coaps and blockwise 
    (missing transfer of scheme, ends up in a lot of "intermediary block" requests)

Reply:
1. I see that in the javadoc, but does we plan to reuse it ?

2. I understand the more we run code, the more we have chance to detect issue but this is not necessary a good argument to keep that tests on this form. I mean the fact we do not detect this error before means that we should maybe improve our Junit tests instead of counting on plugchecker ?

Anyway, if I well understand :
- plug-client seems to check if a server could be use for checking a client.
- plug-check seems to be used to check a server. 
- plug-server seems to be used to check a client.
(names are strange :p)

So I look at the code of plug-check, especially tests which failed about this line. I'm not sure checking if the block 2 option is present or not is a good way to check we do a block transfer correctly. 
So if we want to "fix" plugchecker, I prefer  :
 a. we just remove the block2 test options ... this is not really less precise than before.
 b. if we really want to check block transfer we need to do the test without transparent block mode.

Reply:
> 1. I see that in the javadoc, but does we plan to reuse it ?

The eclipse sandbox offers exactly that "plugtest". 

Currently the plugtest client is used.
> Dec 2017:
> 334 different hosts, 2157 tests
> Jan 2018 (until now):
> 213 different hosts, 1155 tests 

But FMPOV, both plugtest clients are very useful.
There are other possibilities for the two remaining issues:
- block2 option in transparent response
- notify in in `Request.waitForResponse`

As said above, I chose these fixes to keep the efforts limited. But If you feel, that's a wrong priority, I would go for adjust the plugchecker in a first step to support coaps and postpone the fixes.

> that we should maybe improve our Junit tests instead of counting on plugchecker ?

From my personal experience, I disagree. I don't see, that the unit test have a constructed coverage and I don't see, that a constructed coverage is possible with the current number of committers. So I would simple prefer to provide "high level test tools", as the plugchecker, to get aware of issues, which then could be put into a unit test (as I did for the coaps blockwise in #534).

Reply:
JUnit could be used for High level tests, but ok even if I'm not really convinced, let's keep plug projects.

I would prefer to not return the block2 option for transparent block mode, if you agree.
So what do we do ? a. or b. or something else ?

Reply:
Currently I isolate the changes from the "fixes". For the notifies I add a NotificationListener, which forwards the response again to the origin request, but just for these test. For the blockwise I'm searching for an alternative solution.